test/components/tree-view.spec.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import { expect, ClientRenderer, sinon, simulate, waitFor } from 'test-drive-react';
+import { TreeView, TreeItem } from '../../src';
+import { TreeViewDemo, treeData } from '../../demo/components/tree-view-demo';
+import { TreeItemData } from '../../src/components/tree-view/tree-view';
+
+const treeView = 'TREE_VIEW';
+const treeItem = 'TREE_ITEM';
+
+const noop = (item: Object) => false;
+
+function getLabelsList(data: {label: string, children?: Object[]}): string[] {
+    return [data.label]
+               .concat(...(data.children || [])
+               .map(getLabelsList));
+}
+
+function getAllNodeLabels(treeData: Object[]) {
+    return treeData.map(getLabelsList).reduce((prev, next) => [...prev, ...next]);
+}
+
+describe('<TreeView />', () => {
+    const clientRenderer = new ClientRenderer();
+    afterEach(() => clientRenderer.cleanup());
+
+    function getTreeItem(id: string) {
+        return `${treeItem}_${id.replace(' ', '_')}`;
+    }
+
+    it('renders a tree view with a few children, clicks ones of then', async () => {
+        const { select, waitForDom } = clientRenderer.render(<TreeViewDemo />);
+
+        await waitForDom(() => expect(select(treeView + '_DEMO')).to.be.present());
+
+        const allNodesLabels = getAllNodeLabels(treeData);
+        await waitForDom(() => allNodesLabels.forEach(item =>
+            expect(select(treeView + '_DEMO', getTreeItem(item)), `${item} was not present`).to.be.present()));
+
+        const elementToSelect = select(treeView + '_DEMO', getTreeItem(allNodesLabels[2]));
+
+        simulate.click(elementToSelect);
+
+        return waitForDom(() => expect(elementToSelect).to.have.attr('data-selected', 'true'));
+
+    });
+
+    describe('Using default renderer', () => {
+        it('renders correct children', () => {
+            const { select, waitForDom } = clientRenderer.render(<TreeView dataSource={treeData} />);
+
+            return waitForDom(() =>
+                treeData.forEach((item: TreeItemData) =>
+                    expect(select(treeView, getTreeItem(item.label)), `${item.label} was not present`).to.be.present()));
+        });
+
+        it('invokes the onSelectItem callback when an item is clicked', () => {
+            const onSelectItem = sinon.spy();
+            const { select } = clientRenderer.render(<TreeView dataSource={treeData} onSelectItem={onSelectItem}/>);
+
+            const firstNodeChildren = treeData[0].children;
+
+            const nodeLabelToClick = getTreeItem(firstNodeChildren![1].label);
+
+            simulate.click(select(nodeLabelToClick));
+
+            return waitFor(() => {
+                expect(onSelectItem).to.have.been.calledOnce;
```

Comment:
i actually check that as well (with the assert that the clicked element got the data-selected=true attirbute